packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -134,11 +134,25 @@ export class Bootstrapper {
         params.createCustomerMasterKey === false || currentKmsKeyId === undefined ? USE_AWS_MANAGED_KEY :
           undefined);
 
+    let kmsEcrKeyId = undefined;
+
+    if (params.ecrKey) {
+      // key is given
+      kmsEcrKeyId = params.ecrKey;
+    } else if (params.ecrKey === CREATE_NEW_KEY) {
```

Comment:
As the CREATE_NEW_KEY constant resolves in an empty string, the branch would not be ignored. But I understand your point and made it more self explaining by replacing CREATE_NEW_KEY with an explicit empty string.

Reply:
But otherwise, the behaviour is ok from my point of view. If there was a key given, use it, otherwise create a new. 
Splitting this up into two different (commandline) parameters would be an alternative, but would that be cool?
It is like the encryption of the assetbucket, so we should not change the expected behaviour.
And we cannot change both because of the backwards compatibility.